src/client/pages/Payment/TrustlyIframe.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useEffect, ReactEventHandler } from 'react'
+import styled from '@emotion/styled'
+import { SUCCESS_SUFFIX, FAIL_SUFFIX } from './Payment.constants'
+
+type TrustlyIframeProps = {
+  url: string
+  onSuccess: () => void
+  onFail: () => void
+}
+
+export const TrustlyIframe = (props: TrustlyIframeProps) => {
+  const { url, onSuccess, onFail } = props
+
+  useEffect(() => {
+    const handler = (event: MessageEvent) => {
+      const trustlyOrigin = new URL(url).origin
+      if (event.origin !== trustlyOrigin) return
+
+      const message = JSON.parse(event.data)
+      if (message.method === 'OPEN_APP') {
+        window.location.assign(message.appURL)
+      }
+    }
+
+    window.addEventListener('message', handler)
+    return () => window.removeEventListener('message', handler)
+  }, [url])
+
+  const handleLoad: ReactEventHandler<HTMLIFrameElement> = (event) => {
+    const href = event.currentTarget.contentWindow?.location.href
+    if (href?.endsWith(SUCCESS_SUFFIX)) {
```

Comment:
Should we consider the eventual existence of query params into trustly iframe URL? 🤔 
I mean, if someday one is present this logic will fail since it's based on `href?.endsWith`. Maybe use `pathname` in this case.

Reply:
right good point, it's been working until now though but perhaps we can take a stab at it anyways